### DISCUSSION TITLE
Configuring the curl plugin with a private CA

### DISCUSSION THREAD

**Discussion by MrMarlin** [strongswan/strongswan#1764] [Q&A] [answered]
Hi strongSwan community

I am currently setting up a vpn endpoint which uses certificate based authentication. The certificates are issued via an internal Hashicorp Vault secret engine. To check the certificates validity, we want to use the CRL endpoint of vault, which is served via HTTPS. 

Unfortunately this does not work, because the CRL endpoint has a self signed certificate which the curl plugin can not verify. I also can not add the private CA to the installed CAs because on FreeBSD curl is hardcoded to Mozillas CA bundle which is installed via a package. Is there a possibility to tell the curl plugin which CA bundle to use?

Cheers

**Comment by tobiasbrunner** [comment, marked as answer]
No, that's currently not possible.  I guess it would require setting `CURLINFO_CAINFO`. As a global option that could be added relatively easily.

But as mentioned via IRC, it's usually not necessary to protect fetching the CRL via HTTPS.

**Reply by MrMarlin** [reply]
Thanks for the reply. In our setup with vault it is not feasible to have an extra endpoint for non HTTPS requests. Then it's easier to fetch the CRL via cronjob and use a file URI in strongswan. But it would have been a cleaner solution, to fetch the CRL directly via URL in strongswan.
